File: src/yetty/platform/ios/platform-paths.m
/* iOS platform paths - app container directories */

#import <Foundation/Foundation.h>
#include <stdio.h>

static char cache_dir_buf[512];
static char config_dir_buf[512];

const char *yetty_platform_get_cache_dir(void)
{
    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES);
    if (paths.count > 0) {
        snprintf(cache_dir_buf, sizeof(cache_dir_buf), "%s/yetty", [paths[0] UTF8String]);
        return cache_dir_buf;
    }
    return "/tmp/yetty";
}

const char *yetty_platform_get_runtime_dir(void)
{
    return yetty_platform_get_cache_dir();
}

const char *yetty_platform_get_config_dir(void)
{
    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSApplicationSupportDirectory, NSUserDomainMask, YES);
    if (paths.count > 0) {
        snprintf(config_dir_buf, sizeof(config_dir_buf), "%s/yetty", [paths[0] UTF8String]);
        return config_dir_buf;
    }
    return "/tmp/yetty";
}

File: src/yetty/yplatform/windows/platform-paths.c
/* Windows platform paths - AppData directories */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

static char cache_dir_buf[512];
static char data_dir_buf[512];
static char runtime_dir_buf[512];
static char config_dir_buf[512];

const char *yetty_platform_get_cache_dir(void)
{
    const char *localAppData = getenv("LOCALAPPDATA");
    if (localAppData) {
        snprintf(cache_dir_buf, sizeof(cache_dir_buf), "%s\\yetty\\cache", localAppData);
        return cache_dir_buf;
    }
    return "C:\\temp\\yetty";
}

const char *yetty_platform_get_data_dir(void)
{
    const char *localAppData = getenv("LOCALAPPDATA");
    if (localAppData) {
        snprintf(data_dir_buf, sizeof(data_dir_buf), "%s\\yetty\\data", localAppData);
        return data_dir_buf;
    }
    return "C:\\temp\\yetty";
}

const char *yetty_platform_get_runtime_dir(void)
{
    const char *temp = getenv("TEMP");
    if (temp) {
        snprintf(runtime_dir_buf, sizeof(runtime_dir_buf), "%s\\yetty", temp);
        return runtime_dir_buf;
    }
    return "C:\\temp\\yetty";
}

const char *yetty_platform_get_config_dir(void)
{
    const char *appData = getenv("APPDATA");
    if (appData) {
        snprintf(config_dir_buf, sizeof(config_dir_buf), "%s\\yetty", appData);
        return config_dir_buf;
    }
    return "C:\\temp\\yetty";
}
